actors/miner/tests/miner_actor_test_precommit_batch.rs

```diff
@@ -0,0 +1,320 @@
+        assert_simple_batch(
+            3,
+            TokenAmount::zero(),
+            TokenAmount::zero(),
+            &[DealSpec { size: 32 << 30, verified_size: 0, ids: vec![1] }],
+            ExitCode::OK,
+            "",
+        );
+    }
+
+    #[test]
+    fn many_deals() {
+        assert_simple_batch(
+            3,
+            TokenAmount::zero(),
+            TokenAmount::zero(),
+            &[
+                DealSpec { size: 32 << 30, verified_size: 0, ids: vec![1] },
+                DealSpec { size: 0, verified_size: 32 << 30, ids: vec![1] },
+                DealSpec { size: 16 << 30, verified_size: 16 << 30, ids: vec![1, 2] },
+            ],
+            ExitCode::OK,
+            "",
+        );
+    }
+
+    #[test]
+    fn empty_batch() {
+        assert_simple_batch(
+            0,
+            TokenAmount::zero(),
+            TokenAmount::zero(),
+            &[],
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            "batch empty",
+        );
+    }
+
+    #[test]
+    fn too_many_sectors() {
+        assert_simple_batch(
+            Policy::default().pre_commit_sector_batch_max_size + 1,
+            TokenAmount::zero(),
+            TokenAmount::zero(),
+            &[],
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            "batch of 257 too large",
+        );
+    }
+
+    #[test]
+    fn insufficient_balance() {
+        assert_simple_batch(
+            10,
+            TokenAmount::from(-1),
+            TokenAmount::zero(),
+            &[],
+            ExitCode::USR_INSUFFICIENT_FUNDS,
+            "insufficient funds",
+        );
+    }
+
+    #[test]
+    fn one_bad_apple_ruins_batch() {
+        // This test does not enumerate all the individual conditions that could cause a single precommit
+        // to be rejected. Those are covered in the PreCommitSector tests, and we know that that
+        // method is implemented in terms of a batch of one.
+
+        let period_offset = ChainEpoch::from(100);
+
+        let h = ActorHarness::new(period_offset);
+        let mut rt = h.new_runtime();
+        rt.set_balance(TokenAmount::from(BIG_BALANCE));
+        rt.set_received(TokenAmount::zero());
+
+        let precommit_epoch = period_offset + 1;
+        rt.set_epoch(precommit_epoch);
+        h.construct_and_verify(&mut rt);
+        let dl_info = h.deadline(&rt);
+
+        let sector_expiration =
+            dl_info.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period;
+        let sectors = vec![
+            h.make_pre_commit_params(100, precommit_epoch - 1, sector_expiration, vec![]),
+            h.make_pre_commit_params(101, precommit_epoch - 1, sector_expiration, vec![]),
+            h.make_pre_commit_params(102, precommit_epoch - 1, rt.epoch, vec![]), // Expires too soon
+        ];
+
+        expect_abort_contains_message(
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            "sector expiration",
+            h.pre_commit_sector_batch(
```

Comment:
nit: rt.reset() after this call too.  This one is more for explanatory clarity than possible footguns like the one above but its probably better to have it.